rust/agents/relayer/src/msg/pending_message.rs

```diff
@@ -0,0 +1,349 @@
+use std::fmt::{Debug, Formatter};
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+use async_trait::async_trait;
+use derive_new::new;
+use eyre::{Context, Result};
+use prometheus::{IntCounter, IntGauge};
+use tracing::{debug, error, info, instrument, warn};
+
+use hyperlane_base::db::HyperlaneDB;
+use hyperlane_base::{CachingMailbox, CoreMetrics};
+use hyperlane_core::{HyperlaneChain, HyperlaneDomain, HyperlaneMessage, Mailbox, U256};
+
+use super::{
+    gas_payment::GasPaymentEnforcer,
+    metadata::{BaseMetadataBuilder, MetadataBuilder},
+    pending_operation::*,
+};
+
+/// The message context contains the links needed to submit a message. Each
+/// instance is for a unique origin -> destination pairing.
+pub struct MessageCtx {
```

Comment:
On the fence for this one because it is `pub(crate)` and not `pub` (I know it looks like it is `pub` but in the parent module it has a limited export.